### ISSUE TITLE
Could we please have class docs?

### ISSUE THREAD

**Issue by skills-up** [amabnl/amadeus-ws-client#272]
Hi,

I know this may sound unreasonable, but is it possible to have class docs (all classes and their methods) in one place, specially with their purpose (when to use each of the options)?

Samples do provide useful insights for some common use cases, but having the wholistic docs in one place will be really useful. As of now, I've to repeatedly dig into the code to see which all options are permitted, and what each of them entail. But, it is still better than digging through Amadeus docs.

Hope this request is found useful for your consideration.

Thanks,
Gaurav

**Comment by DerMika** [comment]
I'd like to do something like that, but it'd have to be deployed using something like gh-pages. There is probably a way to do this with the TravisCI build, but I'd have to investigate that. 

If anyone has a better idea, let me know!

That said, you can never expect this library to be complete or up-to-date with Amadeus' Extranet. You'll always need that, and I don't plan to become the source of truth for Amadeus Web Services.

**Comment by therealartz** [comment]
@DerMika I will not say that this idea is better, but as a possible option we could generate API documentation from PHPDoc blocks.

**Comment by DerMika** [comment]
Yes, that was actually my plan, like you say, somewhere in the travisCI build perhaps, and publish it to GH-Pages somehow.. Haven't done that before though, so I'd need to to some research.
